ape_hardhat/providers.py

```diff
@@ -302,9 +302,7 @@ def set_block_gas_limit(self, gas_limit: int) -> bool:
         return self._make_request("evm_setBlockGasLimit", [hex(gas_limit)]) is True
 
     def set_timestamp(self, new_timestamp: int):
-        pending_timestamp = self.get_block("pending").timestamp
-        seconds_to_increase = new_timestamp - pending_timestamp
-        self._make_request("evm_increaseTime", [seconds_to_increase])
+        self._make_request("evm_setNextBlockTimestamp", [new_timestamp])
```

Comment:
If you look at `test_set_timestamp`, you will notice the test gives wiggle room because of the flakiness of increasing seconds.

I believe now we can adjust the assertion to be more exact. Thank you!

if you want to improve that test, it will be greatly appreciated

Reply:
I went ahead and did it!